packages/botonic-api/src/rest/utils/paginator.ts

```diff
@@ -0,0 +1,59 @@
+import { Request } from 'express'
+
+export const FIRST_PAGE = 1
+export const DEFAULT_PAGE_SIZE = 10
+export const MAX_PAGE_SIZE = 100
+
+type JsonResponse<T> = {
+  previous?: string
+  next?: string
+  count: number
+  results: T[]
+}
+
+export class Paginator {
+  private readonly baseUrl: string
+  private readonly page: number
+  private readonly pageSize: number
+
+  constructor(
+    request: Request,
+    page = FIRST_PAGE,
+    pageSize = DEFAULT_PAGE_SIZE
+  ) {
+    this.baseUrl = this.getBaseUrl(request)
+    this.page = page < FIRST_PAGE ? FIRST_PAGE : page
+    this.pageSize = pageSize > MAX_PAGE_SIZE ? MAX_PAGE_SIZE : pageSize
```

Comment:
are you considering negative values here?

Reply:
No, I'm forcing always the `page` to be at least 1 (`FIRST_PAGE`) and the `pageSize` to be as maximum as `MAX_PAGE_SIZE`.
Is this what you're asking?

Reply:
Good observation here

Reply:
yes, I meant for negative `pageSize` values.

Reply:
Yes, good point! 
I've added a minimum value of 1 for `pageSize`.